python/python/lance/torch/__init__.py

```diff
@@ -17,7 +17,7 @@
 try:
     import torch
 except ImportError:
-    raise ImportError("PyTorch is not installed")
+    raise ImportError("PyTorch>=2.0 is not installed")
```

Comment:
1. Are we actually checking for >=2 and 2. do we want to add an `extras` entry to for `pytorch` in the `pyproject.toml`?

Reply:
I am a bit hesitant to put pytorch as an extra dependency, because there are multiple options to install pytorch differently.

https://pytorch.org/get-started/locally/

It is hard for us to get one pytorch dependency correct in our `pyproject.toml`.